### PR TITLE
[parser] Disallow numeric separators in legacy octal like integers

### PR DESCRIPTION
**Pull request by gonzarodriguezt**
| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          | Yes, Spec compliance 👓
| Major: Breaking Change?  | No
| Tests Added + Pass?      | Yes
| Any Dependency Changes?  | No
| License                  | MIT

The follow code should throw an error if the numeric separator plugin is enabled:

**Input Code**

- REPL: https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=AzD6yA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=true&presets=env&prettier=false&targets=&version=7.6.2&externalPlugins=%40babel%2Fplugin-syntax-numeric-separator%407.2.0%2C%40babel%2Fplugin-syntax-bigint%407.4.4

```js
00_0
```

**Expected behavior/code**
It should throw Invalid `Invalid numeric separator after leading 0` error.

The Legacy Octal Like Decimal is defined as:

```
LegacyOctalLikeDecimalIntegerLiteral
0 OctalDigit
LegacyOctalLikeDecimalIntegerLiteral OctalDigit

OctalDigit
0 1 2 3 4 5 6 7
```
[Reference](https://github.com/tc39/test262/blob/master/test/language/literals/numeric/numeric-separators/numeric-separator-literal-lol-00-err.js)

### CODE REVIEW

File: packages/babel-parser/src/tokenizer/index.js [hunk 1]
```diff
@@ -1035,8 +1035,11 @@ export default class Tokenizer extends LocationParser {
       next = this.input.charCodeAt(this.state.pos);
     }
 
-    // disallow numeric separators in non octal decimals
-    if (this.hasPlugin("numericSeparator") && isNonOctalDecimalInt) {
+    // disallow numeric separators in non octal decimals and legacy octal likes
+    if (
+      (this.hasPlugin("numericSeparator") && isNonOctalDecimalInt) ||
+      (this.hasPlugin("numericSeparator") && octal)
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/src/tokenizer/index.js]: Can we check `this.hasPlugin` just once?

```js
this.hasPlugin("numericSeparator") && (octal || isNonOctalDecimalInt)
```

File: packages/babel-parser/test/fixtures/experimental/numeric-separator/invalid-lol/input.js [hunk 2]
```diff
@@ -0,0 +1 @@
+00_0
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/test/fixtures/experimental/numeric-separator/invalid-lol/input.js]: I know that you copied "lol" from test262, but please call it `invalid-legacy-octal-literal`. It took two minutes to me to understand what lol was :laughing:

Comment (gonzarodriguezt) [reply packages/babel-parser/test/fixtures/experimental/numeric-separator/invalid-lol/input.js]: Aww i like lol more 🤣but's a reasonably change... committing in a minute! 👌

File: packages/babel-parser/src/tokenizer/index.js (hunk 1 above)
Comment (gonzarodriguezt) [reply packages/babel-parser/src/tokenizer/index.js]: Oh sure! That's better, thank you!

**Comment by gonzarodriguezt** [comment]
Thanks to you! Super glad to contribute :)

**Pull request merged**